---
title: "Differential DoGs analysis"
author: "Aimin Yan"
output: 
  html_document: 
    fig_caption: no
#    number_sections: yes
---

#This two different Dogs are not differential DoGs

![Dogs](/Users/axy148/Desktop/Screen Shot 2016-08-15 at 12.09.38 PM.png)

# Definition of DoGs 

![DefinitionDogs](/Users/axy148/Desktop/DefintionDogs.png)

+ DoGs are a new transcript type generated by readthrough transcription

+ DoGs are chromatin-bound transcripts highly inducible by osmotic stress

+ DoGs are induced downstream of more than 10% of human protein-coding genes
 
+ DoG induction is mediated through IP3 receptor signaling

+ Being long (often >45 kb) 

+ Being diverse (>2,000 species)

+ DoGs contribute significantly to the human transcriptome.


<!-- ```{r starting_time,eval=TRUE,message=F, warning=F,results='asis',echo=FALSE} -->
<!-- time.string = gsub(":", "-", gsub(" ", "_", as.character(Sys.time()))) -->
<!-- cat("This report is generated at:", time.string,"\n") -->
<!-- ``` -->

# Transcript downstream analysis(Borrow some idea from DoGs, but focusing on the 4.5kb  or 5kb downstream region of transcripts)

```{r add_library, message=F, warning=F,results='hide',echo=F} 
## Load required library
#suppressPackageStartupMessages(library(DESeq2))
#library(ThreeUTR)
#library(hwriter)
#library(knitr)
#library(printr)
```
## Transcript based 

![Txs.gene](/Volumes/Bioinformatics$/2016/Ramin_azhang/for_bioinfo_core/RNA_seq/Results/Num.txs.4.gene.png)

<!-- ```{r input: dir and files, eval=F, message=F, warning=F,results='hide',echo=F} -->
<!-- #dir.name="/media/H_driver/2016/Ramin_azhang/for_bioinfo_core/RNA_seq/Results/" -->
<!-- dir.name="/media/H_driver/2016/Ramin_azhang/for_bioinfo_core/RNA_seq/Results4NewData/" -->
<!-- input.file.pattern="*downstream.count.hg19.strand.based.txt" -->
<!-- #input.file.pattern.upstream="*downstream.count.hg19.txt" -->
<!-- input.file.pattern.upstream="*downstream.count.hg19.strand.based.2.txt" -->
<!-- ``` -->

##Get transcript downstream count

![Count](/Volumes/Bioinformatics$/2016/Ramin_azhang/for_bioinfo_core/RNA_seq/Results/uc002qwk.3.downstream.4500bp.right.2.png)

+ Count with artifacts
![Tcount1](/Users/axy148/Desktop/Screen Shot 2016-08-15 at 1.48.10 PM.png)

+ Count removing artifacts
![Tcount2](/Users/axy148/Desktop/Screen Shot 2016-08-15 at 1.48.53 PM.png)

#DE test for transcript based analysis:

"There is some confusion in the answers to this question that hopefully I can clarify with the three comments below:

1. kallisto produces estimates of transcript level counts, and therefore to obtain an estimate of the number of reads from a gene the correct thing to do is to sum the estimated counts from the constituent transcripts of that gene. Of note in the language above is the word "estimate", which is necessary because in many cases reads cannot be mapped uniquely to genes. However insofar as obtaining a good estimate, the approach of kallisto (and before it Cufflinks, RSEM, eXpress and other "transcript level quantification tools") is superior to na√Øve "counting" approaches for estimating the number of reads originating from a gene. This point has been argued in many papers; among my own papers it is most clearly explained and demonstrated in Trapnell et al.  2013. 

2. Although estimated counts for a gene can be obtained by summing the estimated counts of the constituent transcripts from tools such as kallisto, and the resulting numbers can be rounded to produce integers that are of the correct format for tools such as DESeq, the numbers produced by such an approach do not satisfy the distributional assumptions made in DESeq and related tools. For example, in DESeq2, counts are modeled "as following a negative binomial distribution". This assumption is not valid when summing estimated counts of transcripts to obtain gene level counts, hence the justified concern of Michael Love that plugging in sums of estimated transcript counts could be problematic for DESeq2. In fact, even the estimated transcript counts themselves are not negative binomial distributed, and therefore also those are not appropriate for plugging into DESeq2. His concern is equally valid with many other "count based" differential expression tools.

3. Fortunately there is a solution for performing valid statistical testing of differential abundance of individual transcripts, namely the method implemented in sleuth. The approach is described here. To test for differential abundance of genes, one must first address the question of what that means. E.g. is a gene differential if at least one isoform is? or if all the isoforms are? The tests of sleuth are performed at the granularity of transcripts, allowing for downstream analysis that can capture the varied questions that might make biological sense in specific contexts.

In summary, please do not plug in rounded estimates of gene counts from kallisto into DESeq2 and other tools. While it is technically possible, it is not statistically advisable. Instead, you should use tools that make valid distributional assumptions about the estimates."

<!-- ```{r get_count_DE, eval=TRUE, message=F, warning=F,results='hide',echo=TRUE} -->
<!-- #re.rMAT<-ProcessOutputFilesFrom3UTR(dir.name,input.file.pattern) -->
<!-- ``` -->

<!-- ```{r generate_table, echo=FALSE,results='asis'} -->
<!-- #tbl1<-re.rMAT$DE[1:3,] -->
<!-- #tbl1<-re.rMAT$DE[which(rownames(re.rMAT$DE)=="uc002qwk.3"),] -->
<!-- #knitr::kable(tbl1, digits = 2, caption = 'Gene downstream count table') -->
<!-- ``` -->

<!-- ##Get gene upstream count and test -->
<!-- ```{r get_count_DE2, eval=TRUE, message=F, warning=F,results='hide',echo=TRUE} -->
<!-- #re.rMAT.up<-ProcessOutputFilesFrom3UTR(dir.name,input.file.pattern.upstream) -->
<!-- ``` -->

<!-- ```{r generate_table2, echo=FALSE,results='asis'} -->
<!-- #tbl1<-re.rMAT$DE[1:3,] -->
<!-- #tbl2<-re.rMAT.up$DE[which(rownames(re.rMAT.up$DE)=="uc002qwk.3"),] -->
<!-- #knitr::kable(tbl2, digits = 2, caption = 'Gene upstream count table') -->
<!-- ``` -->

