 1015  ls ~/
 1016  ls ~/IGV_2.3.67/
 1017  cd ~/IGV_2.3.67/
 1018  sh igv.sh 
 1019  ls
 1020  history
 1021  ls ~/Code/Com*
 1022  history >  ~/Code/CommandHistory10.txt
 1023  exit
 1024  ps aux 
 1025  ps aux  | grep igv
 1026  kill 17990
 1027  ls
 1028  cd IGV_2.3.67/
 1029  ls
 1030  sg igv.sh 
 1031  sh igv.sh 
 1032  ls
 1033  history
 1034  history | head
 1035  ls
 1036  ls ~/Code/
 1037  cd ~/Code/
 1038  ls
 1039  ~/Code/ 
 1040  ls ~/
 1041  ls ~/bishopric/
 1042  ls ~/bishopric/RNAseq/
 1043  ls ~/bishopric/RNAseq/rawData/
 1044  ls ~/bishopric/RNAseq/rawData/140128_SN392_0190_AC3EYUACXX/
 1045  less ~/bishopric/RNAseq/rawData/140128_SN392_0190_AC3EYUACXX/merged_counts.txt 
 1046  ls ~/bishopric/RNAseq/rawData/140128_SN392_0190_AC3EYUACXX/ 
 1047  locate Sample_10_raw_count.txt
 1048  less Sample_10_raw_count.txt
 1049  ls ~/bishopric/RNAseq/rawData/140128_SN392_0190_AC3EYUACXX/Sample_10_raw_count.txt
 1050  less ~/bishopric/RNAseq/rawData/140128_SN392_0190_AC3EYUACXX/Sample_10_raw_count.txt
 1051  emacs  ~/Downloads/merge_tables.py 
 1052  emacs ~/Downloads/merge_tables.py > ~/bishopric/RNAseq/rawData/140128_SN392_0190_AC3EYUACXX/sample_information.txt 
 1053  Rscript ~/bishopric/RNAseq/rawData/140128_SN392_0190_AC3EYUACXX/merged_counts.txt > ~/bishopric/RNAseq/rawData/140128_SN392_0190_AC3EYUACXX/sample_information.txt 
 1054  Rscript ~/bishopric/RNAseq/rawData/140128_SN392_0190_AC3EYUACXX/merged_counts.txt
 1055  Rscript Run_DESeq_3.R ~/bishopric/RNAseq/rawData/140128_SN392_0190_AC3EYUACXX/merged_counts.txt
 1056  Rscript Run_DESeq_3.R ~/bishopric/RNAseq/rawData/140128_SN392_0190_AC3EYUACXX/merged_counts.txt > ~/bishopric/RNAseq/rawData/140128_SN392_0190_AC3EYUACXX/sample_information.txt
 1057  emacs ~/bishopric/RNAseq/rawData/140128_SN392_0190_AC3EYUACXX/sample_information.txt
 1058  Rscript Run_DESeq_3.R ~/bishopric/RNAseq/rawData/140128_SN392_0190_AC3EYUACXX/merged_counts.txt /bishopric/RNAseq/rawData/140128_SN392_0190_AC3EYUACXX/sample_information.txt
 1059  Rscript Run_DESeq_3.R ~/bishopric/RNAseq/rawData/140128_SN392_0190_AC3EYUACXX/merged_counts.txt ~/bishopric/RNAseq/rawData/140128_SN392_0190_AC3EYUACXX/sample_information.txt
 1060  emacs ~/bishopric/RNAseq/rawData/140128_SN392_0190_AC3EYUACXX/sample_information.txt
 1061  Rscript Run_DESeq_3.R ~/bishopric/RNAseq/rawData/140128_SN392_0190_AC3EYUACXX/merged_counts.txt ~/bishopric/RNAseq/rawData/140128_SN392_0190_AC3EYUACXX/sample_information.txt
 1062  vi ~/bishopric/RNAseq/rawData/140128_SN392_0190_AC3EYUACXX/sample_information.txt
 1063  Rscript Run_DESeq_3.R ~/bishopric/RNAseq/rawData/140128_SN392_0190_AC3EYUACXX/merged_counts.txt ~/bishopric/RNAseq/rawData/140128_SN392_0190_AC3EYUACXX/sample_information.txt
 1064  Rscript Run_DESeq_3.R ~/bishopric/RNAseq/rawData/140128_SN392_0190_AC3EYUACXX/merged_counts.txt ~/bishopric/RNAseq/rawData/140128_SN392_0190_AC3EYUACXX/sample_information.txt ~/bishopric/RNAseq/rawData/140128_SN392_0190_AC3EYUACXX/output_sample1.csv
 1065  Rscript Run_DESeq_3.R ~/bishopric/RNAseq/rawData/140128_SN392_0190_AC3EYUACXX/merged_counts.txt ~/bishopric/RNAseq/rawData/140128_SN392_0190_AC3EYUACXX/sample_information.txt ~/bishopric/RNAseq/rawData/140128_SN392_0190_AC3EYUACXX/output
 1066  ls ~/bishopric/RNAseq/rawData/140128_SN392_0190_AC3EYUACXX/*.csv
 1067  ls *.py
 1068  less *.py
 1069  less QcSum.py
 1070  Rscript Run_DESeq_3.R ~/bishopric/RNAseq/rawData/140128_SN392_0190_AC3EYUACXX/merged_counts.txt ~/bishopric/RNAseq/rawData/140128_SN392_0190_AC3EYUACXX/sample_information.txt ~/bishopric/RNAseq/rawData/140128_SN392_0190_AC3EYUACXX/output
 1071  ls ~/bishopric/RNAseq/rawData/140128_SN392_0190_AC3EYUACXX/output
 1072  ls ~/bishopric/RNAseq/rawData/140128_SN392_0190_AC3EYUACXX/*.DE.csv
 1073  ls ~/bishopric/RNAseq/rawData/140128_SN392_0190_AC3EYUACXX/*_DE.csv
 1074  ls ~/bishopric/RNAseq/rawData/140128_SN392_0190_AC3EYUACXX/*_sample_info.csv
 1075  ls
 1076  ls ~/bishopric/RNAseq/rawData/140128_SN392_0190_AC3EYUACXX/*_sample_info.csv
 1077  ls ~/bishopric/RNAseq/rawData/140128_SN392_0190_AC3EYUACXX/*.csv
 1078  ls ~/bishopric/RNAseq/rawData/140128_SN392_0190_AC3EYUACXX/*_DE.csv
 1079  rm ~/bishopric/RNAseq/rawData/140128_SN392_0190_AC3EYUACXX/*_DE.csv
 1080  ls ~/bishopric/RNAseq/rawData/140128_SN392_0190_AC3EYUACXX/*_sample_info.csv
 1081  Rscript Run_DESeq_3.R ~/bishopric/RNAseq/rawData/140128_SN392_0190_AC3EYUACXX/merged_counts.txt ~/bishopric/RNAseq/rawData/140128_SN392_0190_AC3EYUACXX/sample_information.txt ~/bishopric/RNAseq/rawData/140128_SN392_0190_AC3EYUACXX/output
 1082  ls ~/bishopric/RNAseq/rawData/140128_SN392_0190_AC3EYUACXX/*_sample_info.csv
 1083  rm ~/bishopric/RNAseq/rawData/140128_SN392_0190_AC3EYUACXX/*_sample_info.csv
 1084  rm ~/bishopric/RNAseq/rawData/140128_SN392_0190_AC3EYUACXX/*_DE.csv
 1085  Rscript Run_DESeq_3.R ~/bishopric/RNAseq/rawData/140128_SN392_0190_AC3EYUACXX/merged_counts.txt ~/bishopric/RNAseq/rawData/140128_SN392_0190_AC3EYUACXX/sample_information.txt ~/bishopric/RNAseq/rawData/140128_SN392_0190_AC3EYUACXX/output
 1086  rm ~/bishopric/RNAseq/rawData/140128_SN392_0190_AC3EYUACXX/*_DE.csv
 1087  Rscript Run_DESeq_3.R ~/bishopric/RNAseq/rawData/140128_SN392_0190_AC3EYUACXX/merged_counts.txt ~/bishopric/RNAseq/rawData/140128_SN392_0190_AC3EYUACXX/sample_information.txt ~/bishopric/RNAseq/rawData/140128_SN392_0190_AC3EYUACXX/output
 1088  ls ~/bishopric/RNAseq/rawData/140128_SN392_0190_AC3EYUACXX/*_DE.csv
 1089  ls ~/bishopric/RNAseq/rawData/140128_SN392_0190_AC3EYUACXX/*.csv
 1090  ls ~/bishopric/RNAseq/rawData/140128_SN392_0190_AC3EYUACXX/*info.csv
 1091  rm ~/bishopric/RNAseq/rawData/140128_SN392_0190_AC3EYUACXX/*info.csv
 1092  ls ~/bishopric/RNAseq/rawData/140128_SN392_0190_AC3EYUACXX/*_DE.csv
 1093  rm ~/bishopric/RNAseq/rawData/140128_SN392_0190_AC3EYUACXX/*_DE.csv
 1094  ls ~/bishopric/RNAseq/rawData/140128_SN392_0190_AC3EYUACXX/*.csv
 1095  ls ~/bishopric/RNAseq/rawData/140128_SN392_0190_AC3EYUACXX/out**.csv
 1096  rm ~/bishopric/RNAseq/rawData/140128_SN392_0190_AC3EYUACXX/out**.csv
 1097  ls ~/bishopric/RNAseq/rawData/140128_SN392_0190_AC3EYUACXX/out**.csv
 1098  Rscript Run_DESeq_3.R ~/bishopric/RNAseq/rawData/140128_SN392_0190_AC3EYUACXX/merged_counts.txt ~/bishopric/RNAseq/rawData/140128_SN392_0190_AC3EYUACXX/sample_information.txt ~/bishopric/RNAseq/rawData/140128_SN392_0190_AC3EYUACXX/output
 1099  ls ~/bishopric/RNAseq/rawData/140128_SN392_0190_AC3EYUACXX/out**.csv
 1100  ls -alh ~/bishopric/RNAseq/rawData/140128_SN392_0190_AC3EYUACXX/out**.csv
 1101  ls ~/bishopric/RNAseq/rawData/140128_SN392_0190_AC3EYUACXX/out**.csv
 1102  rm ~/bishopric/RNAseq/rawData/140128_SN392_0190_AC3EYUACXX/out**.csv
 1103  Rscript Run_DESeq_3.R ~/bishopric/RNAseq/rawData/140128_SN392_0190_AC3EYUACXX/merged_counts.txt ~/bishopric/RNAseq/rawData/140128_SN392_0190_AC3EYUACXX/sample_information.txt ~/bishopric/RNAseq/rawData/140128_SN392_0190_AC3EYUACXX/output
 1104  ls ~/bishopric/RNAseq/rawData/140128_SN392_0190_AC3EYUACXX/out**.csv
 1105  rm ~/bishopric/RNAseq/rawData/140128_SN392_0190_AC3EYUACXX/out**.csv
 1106  Rscript Run_DESeq_3.R ~/bishopric/RNAseq/rawData/140128_SN392_0190_AC3EYUACXX/merged_counts.txt ~/bishopric/RNAseq/rawData/140128_SN392_0190_AC3EYUACXX/sample_information.txt ~/bishopric/RNAseq/rawData/140128_SN392_0190_AC3EYUACXX/output
 1107  ls ~/bishopric/RNAseq/rawData/140128_SN392_0190_AC3EYUACXX/out**.csv
 1108  mkdir R
 1109  mkdir RNA-seq-data-analysis-pipeline
 1110  ls
 1111  git status
 1112  ls *.R
 1113  git status
 1114  ls *.R
 1115  mv Run_DESeq_3.R ./RNA-seq-data-analysis-pipeline/
 1116  git status
 1117  git a
 1118  git status
 1119  git add Run_DESeq2.R
 1120  git commit -m "Update_12_14_2015"
 1121  git pull --rebase
 1122  git staus
 1123  git status
 1124  ls
 1125  git add RNA-seq-data-analysis-pipeline/*
 1126  git commit -m "Add RNA seq data analysis folder"
 1127  git push origin master
 1128  git status
 1129  git add RNA-seq-data-analysis-pipeline/Run_DESeq_3.R
 1130  git commit -m "add usage"
 1131  git pull --rebase
 1132  git push origin master
 1133  git status
 1134  git add RNA-seq-data-analysis-pipeline/Run_DESeq_3.R
 1135  git commit -m "delete some commented lines"
 1136  git pull --rebase
 1137  git push origin master
 1138  ls 
 1139  ls *.R
 1140  mv alignment2count_PE.R ./RNA-seq-data-analysis-pipeline/ 
 1141  mv alignment2count_SE.R ./RNA-seq-data-analysis-pipeline/ 
 1142  mv run_trimmomatic_PE.R ./RNA-seq-data-analysis-pipeline/ 
 1143  mv run_trimmomatic_SE.R ./RNA-seq-data-analysis-pipeline/ 
 1144  ls ./RNA-seq-data-analysis-pipeline/
 1145  git status
 1146  cd RNA-seq-data-analysis-pipeline/
 1147  git status
 1148  git add alignment2count_PE.R 
 1149  git commit -m "run alignment to get count"
 1150  git add alignment2count_SE.R
 1151  git commit -m "run alignment to get count for single end"
 1152  git add run_trimmomatic_PE.R
 1153  git commit -m "run trimmomatic to trim short read file"
 1154  git add run_trimmomatic_SE.R
 1155  git commit -m "run trimmomatic to trim short read file for single end"
 1156  git pull --rebase
 1157  git remote -v
 1158  cd ..
 1159  git status
 1160  git rm alignment2count_PE.R
 1161  git rm alignment2count_SE.R
 1162  git rm run_trimmomatic_PE.R
 1163  git rm run_trimmomatic_SE.R
 1164  git pull --rebase
 1165  git status
 1166  git commit -m "delete unnecessary R code from code"
 1167  git pull --rebase
 1168  git push origin master
 1169  ls
 1170  ls *.rd
 1171  ls READ*
 1172  less README.txt 
 1173  cp README.md ./RNA-seq-data-analysis-pipeline/
 1174  git add ./R
 1175  git add ./RNA-seq-data-analysis-pipeline/README.md
 1176  git commit -m "add readme file for RNA seq data analysis"
 1177  git pull --rebase
 1178  git push origin master
 1179  ls
 1180  cd RNA-seq-data-analysis-pipeline/
 1181  ls
 1182  emacs README.md 
 1183  git status
 1184  git add README.md
 1185  git commit -m "Update readme file"
 1186  git pull --rebase
 1187  git push origin master
 1188  emacs README.md 
 1189  git add README.md
 1190  git commit -m "Update readme file"
 1191  git pull --rebase
 1192  git push origin master
 1193  emacs README.md 
 1194  git add README.md
 1195  git commit -m "Update readme file"
 1196  git pull --rebase
 1197  git push origin master
 1198  emacs README.md 
 1199  git add README.md
 1200  git commit -m "Update readme file"
 1201  git pull --rebase
 1202  git push origin master
 1203  git status
 1204  ls
 1205  pwd
 1206  ls
 1207  ls /media/
 1208  ls /media/DATA/
 1209  ls /media/DATA/Koji/
 1210  ls
 1211  ls ../
 1212  ls ../*.py
 1213  cp ../merge_tables.py ../QcSum.py .
 1214  ls
 1215  git add merge_tables.py 
 1216  git commit -m "To merge all count files to a table"
 1217  git add QcSum.py 
 1218  git commit -m "To get a summary from QC files After applying FastQC on fastq files"
 1219  git status
 1220  git pull --rebase
 1221  git push master
 1222  git push origin master
 1223  ls *.sh
 1224  ls ../*sh
 1225  less ../Bash_run_submit_job_4_sorted_bam.sh
 1226  less ../Bash_run_submit_job_2.sh
 1227  less ../Bash_run_submit_job_3.sh
 1228  less ../Bash_run_submit_job_4_sorted_bam_2.sh
 1229  less ../Bash_run_submit_job_4_sorted_bam.sh
 1230  cp ../Bash_run_submit_job_4_sorted_bam.sh .
 1231  cp ../Bash_run_submit_job_4_sorted_bam_2.sh .
 1232  git status
 1233  ls
 1234  git add Bash_run_submit_job_4_sorted_bam.sh
 1235  git commit -m "sort bam files based on gene name"
 1236  git add Bash_run_submit_job_4_sorted_bam_2.sh
 1237  git commit -m "To get mapping count for eahc gene using the sorted bam files"
 1238  git pull --rebase
 1239  git push origin master
 1240  ls ../
 1241  ls ../*.sh
 1242  less RunAlignment2.sh
 1243  less ../RunAlignment2.sh
 1244  less ../Run_Fq_24.sh
 1245  ls ../*.sh
 1246  cp ../RunAlignment4SingleEnd.sh .
 1247  cp RunAlignment4SingleEnd.sh RunAlignment4PairEnd.sh 
 1248  emacs RunAlignment4PairEnd.sh 
 1249  git status
 1250  git add RunAlignment4PairEnd.sh 
 1251  git commit -m "Run tophat2 for pair end short read files"
 1252  git add RunAlignment4SingleEnd.sh 
 1253  git commit -m "Run tophat2 for single end short read files"
 1254  git status
 1255  git pull --rebase
 1256  git push origin master
 1257  git status
 1258  git pull --rebase
 1259  ls *.py
 1260  ls
 1261  emacs merge_tables.py 
 1262  python merge_tables.py 
 1263  emacs merge_tables.py 
 1264  ls ~/
 1265  ls ~/bishopric/
 1266  ls ~/bishopric/RNASeq
 1267  ls ~/bishopric/RNA
 1268  ls ~/bishopric/
 1269  ls ~/bishopric/RNAseq/
 1270  ls ~/bishopric/RNAseq/rawData/
 1271  ls ~/bishopric/RNAseq/rawData/140128_SN392_0190_AC3EYUACXX/
 1272  less ~/bishopric/RNAseq/rawData/140128_SN392_0190_AC3EYUACXX/sample_information.txt
 1273  less ~/bishopric/RNAseq/rawData/140128_SN392_0190_AC3EYUACXX/Sample_10_raw_count_after_sorting.txt
 1274  python merge_tables.py ~/bishopric/RNAseq/rawData/140128_SN392_0190_AC3EYUACXX/Sample_10_raw_count_after_sorting.txt ~/bishopric/RNAseq/rawData/140128_SN392_0190_AC3EYUACXX/Output_count_samples10.txt
 1275  emacs merge_tables.py 
 1276  python merge_tables.py -i ~/bishopric/RNAseq/rawData/140128_SN392_0190_AC3EYUACXX/Sample_10_raw_count_after_sorting.txt -o ~/bishopric/RNAseq/rawData/140128_SN392_0190_AC3EYUACXX/Output_count_samples10.txt
 1277  ls ~/bishopric/RNAseq/rawData/140128_SN392_0190_AC3EYUACXX/Output_count_samples10.txt
 1278  emacs merge_tables.py 
 1279  python merge_tables.py -i ~/bishopric/RNAseq/rawData/140128_SN392_0190_AC3EYUACXX/Sample_10_raw_count_after_sorting.txt -o ~/bishopric/RNAseq/rawData/140128_SN392_0190_AC3EYUACXX/Output_count_samples10.txt
 1280  emacs merge_tables.py 
 1281  python merge_tables.py -i ~/bishopric/RNAseq/rawData/140128_SN392_0190_AC3EYUACXX/Sample_10_raw_count_after_sorting.txt -o ~/bishopric/RNAseq/rawData/140128_SN392_0190_AC3EYUACXX/Output_count_samples10.txt
 1282  emacs merge_tables.py 
 1283  python merge_tables.py -i ~/bishopric/RNAseq/rawData/140128_SN392_0190_AC3EYUACXX/Sample_10_raw_count_after_sorting.txt -o ~/bishopric/RNAseq/rawData/140128_SN392_0190_AC3EYUACXX/Output_count_samples10.txt
 1284  emacs merge_tables.py 
 1285  python merge_tables.py -i ~/bishopric/RNAseq/rawData/140128_SN392_0190_AC3EYUACXX/Sample_10_raw_count_after_sorting.txt -o ~/bishopric/RNAseq/rawData/140128_SN392_0190_AC3EYUACXX/Output_count_samples10.txt
 1286  emacs merge_tables.py 
 1287  python merge_tables.py -i ~/bishopric/RNAseq/rawData/140128_SN392_0190_AC3EYUACXX/Sample_10_raw_count_after_sorting.txt -o ~/bishopric/RNAseq/rawData/140128_SN392_0190_AC3EYUACXX/Output_count_samples10.txt
 1288  emacs merge_tables.py 
 1289  python merge_tables.py -i ~/bishopric/RNAseq/rawData/140128_SN392_0190_AC3EYUACXX/Sample_10_raw_count_after_sorting.txt -o ~/bishopric/RNAseq/rawData/140128_SN392_0190_AC3EYUACXX/Output_count_samples10.txt
 1290  emacs merge_tables.py 
 1291  python merge_tables.py -i ~/bishopric/RNAseq/rawData/140128_SN392_0190_AC3EYUACXX/Sample_10_raw_count_after_sorting.txt -o ~/bishopric/RNAseq/rawData/140128_SN392_0190_AC3EYUACXX/Output_count_samples10.txt
 1292  emacs merge_tables.py 
 1293  python merge_tables.py -i ~/bishopric/RNAseq/rawData/140128_SN392_0190_AC3EYUACXX/Sample_10_raw_count_after_sorting.txt -o ~/bishopric/RNAseq/rawData/140128_SN392_0190_AC3EYUACXX/Output_count_samples10.txt
 1294  emacs merge_tables.py 
 1295  python merge_tables.py -i ~/bishopric/RNAseq/rawData/140128_SN392_0190_AC3EYUACXX/Sample_10_raw_count_after_sorting.txt -o ~/bishopric/RNAseq/rawData/140128_SN392_0190_AC3EYUACXX/Output_count_samples10.txt
 1296  emacs merge_tables.py 
 1297  python merge_tables.py -i ~/bishopric/RNAseq/rawData/140128_SN392_0190_AC3EYUACXX/Sample_10_raw_count_after_sorting.txt -o ~/bishopric/RNAseq/rawData/140128_SN392_0190_AC3EYUACXX/Output_count_samples10.txt
 1298  emacs merge_tables.py 
 1299  python merge_tables.py -i ~/bishopric/RNAseq/rawData/140128_SN392_0190_AC3EYUACXX/Sample_10_raw_count_after_sorting.txt -o ~/bishopric/RNAseq/rawData/140128_SN392_0190_AC3EYUACXX/Output_count_samples10.txt
 1300  emacs merge_tables.py 
 1301  git status
 1302  git pull --rebase
 1303  emacs merge_tables.py 
 1304  python merge_tables.py ~/bishopric/RNAseq/rawData/140128_SN392_0190_AC3EYUACXX/Sample_10_raw_count_after_sorting.txt ~/bishopric/RNAseq/rawData/140128_SN392_0190_AC3EYUACXX/Output_count_samples10.txt
 1305  ls ~/bishopric/RNAseq/rawData/140128_SN392_0190_AC3EYUACXX/Output_count_samples10.txt
 1306  less ~/bishopric/RNAseq/rawData/140128_SN392_0190_AC3EYUACXX/Output_count_samples10.txt
 1307  ls ~/bishopric/RNAseq/rawData/140128_SN392_0190_AC3EYUACXX/Output_count_samples10.txt
 1308  vi ~/bishopric/RNAseq/rawData/140128_SN392_0190_AC3EYUACXX/Output_count_samples10.txt
 1309  emacs merge_tables.py 
 1310  ls ~/bishopric/RNAseq/rawData/140128_SN392_0190_AC3EYUACXX/[D
 1311  ls ~/bishopric/RNAseq/rawData/140128_SN392_0190_AC3EYUACXX/
 1312  python merge_tables.py ~/bishopric/RNAseq/rawData/140128_SN392_0190_AC3EYUACXX/Sample_10_raw_count.txt ~/bishopric/RNAseq/rawData/140128_SN392_0190_AC3EYUACXX/Output_count_samples10.txt
 1313  cd ~/bishopric/RNAseq/rawData/140128_SN392_0190_AC3EYUACXX/
 1314  python ~/Code/RNA-seq-data-analysis-pipeline/merge_tables.py Sample_10_raw_count.txt Output_count_samples10.txt
 1315  ls Output_count_samples10.txt 
 1316  less Output_count_samples10.txt 
 1317  emacs ~/Code/RNA-seq-data-analysis-pipeline/merge_tables.py
 1318  python ~/Code/RNA-seq-data-analysis-pipeline/merge_tables.py Sample_10_raw_count.txt Output_count_samples10.txt
 1319  les Output_count_samples10.txt 
 1320  less Output_count_samples10.txt 
 1321  cd ~/Code/RNA-seq-data-analysis-pipeline/
 1322  emacs merge_tables.py 
 1323  git add merge_tables.py
 1324  git commit -m "specify a outfile name from comand argument, and pass to the program"
 1325  git pull --rebase
 1326  git push origin master
 1327  ls
 1328  ls ../
 1329  ls ../*.sh
 1330  grep FastQC ../*.sh
 1331  grep \fast ../*.sh
 1332  grep fast ../*.sh
 1333  ls
 1334  ls ../
 1335  les
 1336  less ~/bishopric/RNAseq/rawData/140128_SN392_0190_AC3EYUACXX/all_mod_scores.csv 
 1337  ls *.py
 1338  less QCSum.py
 1339  less QcSum.py
 1340  emacs QcSum.py
 1341  git add QcSum.py
 1342  git commit -m "Add usage for this program"
 1343  git pull --rebase
 1344  git push origin master
 1345  git pull --rebase
 1346  git push origin master
 1347  ls ../
 1348  less ../Run_Fq_24.sh
 1349  grep fastqc ../*.sh
 1350  grep zip ./*.sh
 1351  fastqc
 1352  pwd
 1353  ls
 1354  ls ~/
 1355  ls ~/bishopric/
 1356  ls ~/bishopric/RNAseq/
 1357  ls
 1358  ls ~/bishopric/RNAseq/rawData/
 1359  ls ~/bishopric/RNAseq/rawData/140128_SN392_0190_AC3EYUACXX/
 1360  ls ~/bishopric/RNAseq/rawData/140128_SN392_0190_AC3EYUACXX/*.sh
 1361  less ~/bishopric/RNAseq/rawData/140128_SN392_0190_AC3EYUACXX/*.sh
 1362  ls ~/bishopric/R
 1363  ls ~/bishopric/RNAseq/rawData/140128_SN392_0190_AC3EYUACXX/
 1364  ls ~/bishopric/RNAseq/rawData/
 1365  ls ~/bishopric/RNAseq/
 1366  ls ~/bishopric/
 1367  ls ~/bishopric
 1368  ls ~/
 1369  locate *.sh
 1370  locate "*.sh"
 1371  locate "*.sh" ~/
 1372  locate "*.sh"
 1373  locate ~/ "*.sh"
 1374  man locate
 1375  find ~/ -iname "*.sh"
 1376  locate fastqc
 1377  ack-grep fastqc
 1378  ack-grep fastqc ~/
 1379  ack-grep fastqc
 1380  ack-grep fastqc ~/Code 
 1381  ls
 1382  ls ~/
 1383  ack-grep fastqc ~/workspace 
 1384  ls /home/aiminyan/workspace/rnaseq/tools/FastQC/fastqc
 1385  ls /home/aiminyan/workspace/rnaseq/tools/FastQC/
 1386  /home/aiminyan/workspace/rnaseq/tools/FastQC/fastqc
 1387  cd /home/aiminyan/workspace/rnaseq/tools/FastQC/fastqc
 1388  ls
 1389  cd /home/aiminyan/workspace/rnaseq/tools/FastQC/
 1390  ls
 1391  ack-grep fastqc ./ 
 1392  ls
 1393  vi run_fastqc.bat 
 1394  cd /home/aiminyan/workspace/rnaseq/tools/FastQC/
 1395  vi run_fastqc.bat 
 1396  ack-grep fastqc ./
 1397  pwd
 1398  ls
 1399  pwd
 1400  emacs /home/aiminyan/.bashrc
 1401  source /home/aiminyan/.bashrc
 1402  ls
 1403  cd ~/
 1404  ls
 1405  cd bishopric/
 1406  ls
 1407  cd RNAseq/
 1408  ls
 1409  cd rawData/
 1410  ls
 1411  cd 140128_SN392_0190_AC3EYUACXX/
 1412  ls
 1413  ls ~/media
 1414  ls /media/DATA/
 1415  ls /media/DATA/nBishopric_Project1_201348189-01_S_1_1.txt
 1416  ls /media/DATA/nBishopric_Project1_201348189-01_S_1_2.txt
 1417  fastqc /media/DATA/nBishopric_Project1_201348189-01_S_1_1.txt /media/DATA/nBishopric_Project1_201348189-01_S_1_2.txt
 1418  ls -lrt
 1419  ls 
 1420  fastqc -o QC_output /media/DATA/nBishopric_Project1_201348189-01_S_1_1.txt
 1421  fastqc /media/DATA/nBishopric_Project1_201348189-01_S_1_1.txt -o QC_output
 1422  mkdir QC_output
 1423  fastqc -o QC_output /media/DATA/nBishopric_Project1_201348189-01_S_1_1.txt
 1424  ls
 1425  ls ./QC_output/
 1426  ls -lrt ./QC_output/
 1427  ls -lrt .
 1428  fastqc ./QC_output /media/DATA/nBishopric_Project1_201348189-01_S_1_1.txt /media/DATA/nBishopric_Project1_201348189-01_S_1_2.txt
 1429  fastqc -o ./QC_output /media/DATA/nBishopric_Project1_201348189-01_S_1_1.txt /media/DATA/nBishopric_Project1_201348189-01_S_1_2.txt
 1430  ls 
 1431  ls -lrt
 1432  ls -lrt QC_output/
 1433  cd QC_output/
 1434  ls
 1435  python ~/Code/RNA-seq-data-analysis-pipeline/QcSum.py
 1436  ls
 1437  less all_mod_scores.csv 
 1438  cd ~/Code/RNA-seq-data-analysis-pipeline/
 1439  ls
 1440  pwd
 1441  ls
 1442  git pull --rebase 
 1443  git push origin master 
 1444  pwd
 1445  ls
 1446  less ofile 
 1447  ls -alh ofile 
 1448  rm ofile 
 1449  ls
 1450  emacs QCSum.py
 1451  ls
 1452  emacs QcSum.py
 1453  ls
 1454  git status
 1455  ls /media/DATA/
 1456  ls ~/
 1457  ls ~/bishopric/
 1458  ls ~/bishopric/RNAseq/
 1459  ls ~/bishopric/RNAseq/rawData/
 1460  ls
 1461  ls ~/bishopric/RNAseq/rawData/
 1462  ls ~/bishopric/RNAseq/rawData/140128_SN392_0190_AC3EYUACXX/
 1463  ls ~/bishopric/RNAseq/rawData/140128_SN392_0190_AC3EYUACXX/QC_output/
 1464  cd ~/bishopric/RNAseq/rawData/140128_SN392_0190_AC3EYUACXX/QC_output/
 1465  python ~/Code/RNA-seq-data-analysis-pipeline/QcSum.py QC_sum_4_2_fs.csv
 1466  emacs /home/aiminyan/Code/RNA-seq-data-analysis-pipeline/QcSum.py
 1467  python ~/Code/RNA-seq-data-analysis-pipeline/QcSum.py QC_sum_4_2_fs.csv
 1468  python ~/Code/RNA-seq-data-analysis-pipeline/QcSum.py
 1469  emacs /home/aiminyan/Code/RNA-seq-data-analysis-pipeline/QcSum.py
 1470  python ~/Code/RNA-seq-data-analysis-pipeline/QcSum.py QC_sum_4_2_fs.csv
 1471  python ~/Code/RNA-seq-data-analysis-pipeline/QcSum.py
 1472  python ~/Code/RNA-seq-data-analysis-pipeline/QcSum.py QC_sum_4_2_fs.csv
 1473  emacs /home/aiminyan/Code/RNA-seq-data-analysis-pipeline/QcSum.py
 1474  python ~/Code/RNA-seq-data-analysis-pipeline/QcSum.py QC_sum_4_2_fs.csv
 1475  ls
 1476  less QC_sum_4_2_fs.csv 
 1477  cd /home/aiminyan/Code/RNA-seq-data-analysis-pipeline/
 1478  ls
 1479  git status
 1480  git add QcSum.py
 1481  git commit -m "specify a QC summary output file name"
 1482  git pull --rebase
 1483  git push origin master 
 1484  git pull --rebase
 1485  git push origin master 
 1486  locate Sample_9.txt
 1487  emacs /home/aiminyan/Result/Jianping/Sample_9.txt
 1488  locate FileBam.txt
 1489  ls ~/Code/CommandHistory
 1490  ls ~/Code/CommandHistory*
 1491  history > ~/Code/CommandHistory11.txt
 1492  exit
 1493  java -version
 1494  sudo apt-get update
 1495  locate source.list
 1496  sudo gedit /etc/apt/sources.list
 1497  sudo apt-get update
 1498  java -version
 1499  sudo apt-get install openjdk-8-jre
 1500  java -version
 1501  sudo update-alternatives --config java
 1502  sudo apt-get install default-jre
 1503  java -version
 1504  sudo update-alternatives --config java
 1505  sudo apt-get install openjdk-7-jre 
 1506  java -version
 1507  sudo update-alternatives --config java
 1508  java -version
 1509  ls ~/Code/CommandHistory*
 1510  history > ~/Code/CommandHistory12.txt
 1511  exit
 1512  ls ~/Code/CommandHistory*
 1513  history >  ~/Code/CommandHistory13.txt
 1514  exit
 1515  java
 1516  java -v
 1517  java -version
 1518  ls
 1519  ls ~/Code/CommandHistory*
 1520  history > ~/Code/CommandHistory14.txt
 1521  exit
 1522  ps aux | grep igv
 1523  kill -9 17854
 1524  ls
 1525  cd IGV_2.3.67/
 1526  ls
 1527  sh igv.sh 
 1528  ls ~/Code/Com*
 1529  ls -l ~/Code/Com*
 1530  history > ~/Code/CommandHistory15.txt
 1531  exit
 1532  ls /media/
 1533  ls /media/DATA/
 1534  ls /media/DATA/*.bed
 1535  ls /media/DATA/Chip_Seq/
 1536  head /media/DATA/Chip_Seq/mtgr1_peaks.bed 
 1537  R
 1538  ssh axy148@pegasus.ccs.miami.edu
 1539  ls ~/Code/Com*
 1540  ls -l ~/Code/Com*
 1541  history > ~/Code/CommandHistory16.txt
 1542  exit
 1543  ls ~/
 1544  ls ~/Bw_Files/
 1545  ls ~/Bw_Files/#.bw
 1546  ls ~/Bw_Files/*3.bw
 1547  ls ~/Code/CommandHistory*
 1548  ls -l ~/Code/CommandHistory*
 1549  history > ~/Code/CommandHistory17.txt
 1550  exit
 1551  ls home/aiminyan/R/x86_64-pc-linux-gnu-library/3.2/ChIPseeker/extdata/GEO_sample_data/GSM1174480_ARmo_0M_peaks.bed.gz
 1552  ls ~/R/x86_64-pc-linux-gnu-library/3.2/ChIPseeker/extdata/GEO_sample_data/GSM1174480_ARmo_0M_peaks.bed.gz
 1553  ls ~/R/x86_64-pc-linux-gnu-library/3.2/ChIPseeker/extdata/GEO_sample_data/
 1554  ls ~/R/x86_64-pc-linux-gnu-library/3.2/ChIPseeker/extdata/
 1555  ls ~/R/x86_64-pc-linux-gnu-library/3.2/ChIPseeker/extdata/GEO_sample_data/sample_peaks.txt
 1556  ls ~/R/x86_64-pc-linux-gnu-library/3.2/ChIPseeker/extdata/GEO_sample_data/
 1557  ls ~/R/x86_64-pc-linux-gnu-library/3.2/ChIPseeker/extdata/
 1558  ls ~/R/x86_64-pc-linux-gnu-library/3.2/ChIPseeker/extdata/sample_peaks.txt 
 1559  less ~/R/x86_64-pc-linux-gnu-library/3.2/ChIPseeker/extdata/sample_peaks.txt 
 1560  less ~/R/x86_64-pc-linux-gnu-library/3.2/ChIPseeker/extdata/GEO_sample_data/ 
 1561  ls ~/R/x86_64-pc-linux-gnu-library/3.2/ChIPseeker/extdata/GEO_sample_data/ 
 1562  cd ~/R/x86_64-pc-linux-gnu-library/3.2/ChIPseeker/extdata/GEO_sample_data/ 
 1563  ls
 1564  zcat GSM1174480_ARmo_0M_peaks.bed.gz | head
 1565  ls
 1566  grep chr7 GSE19013_MCF7_E2-ethl_peaks.bed
 1567  grep chr7 GSE19013_MCF7_E2-ethl_peaks.bed |  wc -l
 1568  cat <(grep chr7 GSE19013_MCF7_E2-ethl_peaks.bed) <(grep chr17 GSE19013_MCF7_E2-ethl_peaks.bed) > GSE19013_MCF7_E2-ethl_chr_7_17_peaks.bed
 1569  ls
 1570  wc -l GSE19013_MCF7_E2-ethl_chr_7_17_peaks.bed
 1571  cat <(grep chr7 GSM1329660_RelA_mid.bed) <(grep chr17 GSM1329660_RelA_mid.bed) > GSM1329660_RelA_mid_chr_7_17_peaks.bed
 1572  ls 
 1573  ls -alh 
 1574  less GSM1329660_RelA_mid_chr_7_17_peaks.bed
 1575  wc -l GSM1329660_RelA_mid_chr_7_17_peaks.bed
 1576  wc -l GSM1329660_RelA_mid_peaks.bed
 1577  wc -l GSM1329660_RelA_mid.bed
 1578  cat <(grep chr7 GSM1329661_RelB_mid.bed) <(grep chr17 GSM1329661_RelB_mid.bed) > GSM1329661_RelB_mid_chr_7_17_peaks.bed
 1579  ls
 1580  ls -alh
 1581  ls /media/DATA/
 1582  ls /media/DATA/Chip_Seq/
 1583  diff GSE19013_MCF7_E2-ethl_peaks.bed  /media/DATA/Chip_Seq/GSE19013_MCF7_E2-ethl_peaks.bed
 1584  diff GSM1329660_RelA_mid.bed  /media/DATA/Chip_Seq/GSM1329660_RelA_mid.bed
 1585  ls
 1586  ls -alh
 1587  rm GSE19013_MCF7_E2-ethl_peaks.bed GSM1329660_RelA_mid.bed GSM1329661_RelB_mid.bed
 1588  ls
 1589  cat <(grep chr7 /media/DATA/Chip_Seq/GSM1329661_RelB_mid.bed) <(grep chr17  /media/DATA/Chip_Seq/GSM1329661_RelB_mid.bed) > GSM1329661_RelB_mid_chr_7_17_peaks.bed 
 1590  ls -alh
 1591  less GSM1329660_RelA_mid_chr_7_17_peaks.bed
 1592  tail GSM1329660_RelA_mid_chr_7_17_peaks.bed
 1593  ls
 1594  head GSM1329661_RelB_mid_chr_7_17_peaks.bed
 1595  tail GSM1329661_RelB_mid_chr_7_17_peaks.bed
 1596  ls
 1597  head GSE19013_MCF7_E2-ethl_chr_7_17_peaks.bed
 1598  tail GSE19013_MCF7_E2-ethl_chr_7_17_peaks.bed
 1599  ls ~/media/DATA
 1600  ls /media/DATA
 1601  ls /media/DATA/Chip_Seq/
 1602  ls ~/
 1603  ls ~/Downloads/
 1604  head ~/Downloads/GSE19013_MCF7_E2-ethl.hpeak.out.txt
 1605  ls ~/Downloads/
 1606  ls ~/Downloads/bedGraphToBigWig 
 1607  ~/Downloads/bedGraphToBigWig 
 1608  chmod 777 ~/Downloads/bedGraphToBigWig 
 1609  ~/Downloads/bedGraphToBigWig 
 1610  ls
 1611  wc -l GSE19013_MCF7_E2-ethl_chr_7_17_peaks.bed 
 1612  wc -l GSM1329660_RelA_mid_chr_7_17_peaks.bed
 1613  ls
 1614  ls ~/Downloads/
 1615  head  ~/Downloads/GSM1329660_RelA.bed
 1616  ls -l ~/Code/Com*
 1617  history > ~/Code/CommandHistory18.txt
 1618  exit
 1619  ls
 1620  pwd
 1621  ls
 1622  ls ~/Code/Co*
 1623  history > ~/Code/CommandHistory19.txt
 1624  exit
 1625  sudo apt-get install mesa-common-dev
 1626  R
 1627  sudo R
 1628  pwd
 1629  history
 1630  ls ~/Code/CommandHistory*
 1631  ls -l ~/Code/CommandHistory*
 1632  history > ~/Code/CommandHistory20.txt
 1633  exit
 1634  ls
 1635  ls igv/
 1636  ls igv/genomes/
 1637  ls igv/genomes/user-defined-genomes.txt 
 1638  less igv/genomes/user-defined-genomes.txt 
 1639  ls igv/genomes/
 1640  ls 
 1641  ls igv
 1642  ls igv/lists/ 
 1643  ls IGV_2.3.67/
 1644  ls igv
 1645  ls igv/genomes/
 1646  ls igv/genomes/hg19.genome 
 1647  ls -alh igv/genomes/hg19.genome 
 1648  head -alh igv/genomes/hg19.genome 
 1649  head  igv/genomes/hg19.genome 
 1650  ssh axy148@pegasus.ccs.miami.edu
 1651  ls
 1652  ls -l ~/Code/CommandHistory*
 1653  history > ~/Code/CommandHistory21.txt
 1654  exit
 1655  ls 
 1656  ls bishopric/  
 1657  ls bishopric/RNAseq/
 1658  ls bishopric/RNAseq/eawData
 1659  ls bishopric/RNAseq/rawData/*.bam
 1660  scp bishopric/RNAseq/rawData/*.bam axy148@pegasus.ccs.miami.edu:~/bishopric/
 1661  ps aux | grep igv
 1662  kill 23703
 1663  ps aux | grep igv
 1664  ps aux | grep igv
 1665  kill 23868
 1666  ps aux | grep igv
 1667  ls
 1668  ls -l ~/Code/CommandHistory*
 1669  history > ~/Code/CommandHistory22.txt
 1670  exit
 1671  sudo  /opt/lampp/lampp start
 1672  ifconfig
 1673  ls ~/Code/CommandHistory*
 1674  history >  ~/Code/CommandHistory23.txt
 1675  exit
 1676  ls
 1677  cd Bw_Files/
 1678  ls
 1679  scp axy148@pegasus.ccs.miami.edu:~/bishopric/*3.bw .
 1680  ls
 1681  cd ~
 1682  ls
 1683  ls -l ~/Code/CommandHistory*
 1684  history > ~/Code/CommandHistory24.txt
 1685  exit
 1686  ps aux | grep igv
 1687  ps aux | grep IGV
 1688  ps aux | grep igv
 1689  kill 22613
 1690  ls
 1691  ls /
 1692  ls /media/
 1693  ls /media/DATA/
 1694  ls -l ~/Code/Com*
 1695  history > ~/Code/CommandHistory25.txt
 1696  exit
 1697  ls
 1698  cd IGV_2.3.67/
 1699  ls
 1700  ls
 1701  sh igv.sh 
 1702  pwd
 1703  ls
 1704  cd ..
 1705  ls
 1706  cd Bw_Files/
 1707  ls
 1708  cd ..
 1709  ls
 1710  cd Code
 1711  ls
 1712  emacs ConvertBamtoBigwig.sh 
 1713  ls -l ~/Comm*
 1714  ls -l ~/Code/C*
 1715  history > ~/Code/CommandHistory26.txt
 1716  exit
 1717  mkdir NGS_tools
 1718  ls
 1719  cd NGS_tools/
 1720  ls
 1721  git clone https://github.com/alexdobin/STAR.git
 1722  cd STAR
 1723  make STAR
 1724  ls
 1725  cd source
 1726  ls
 1727  emacs Makefile 
 1728  cd ..
 1729  ls
 1730  make STAR
 1731  ls
 1732  cd bin/
 1733  ls
 1734  ls Linux_x86_64
 1735  ls Linux_x86_64/STAR
 1736  ./Linux_x86_64/STAR
 1737  cd ../
 1738  ls
 1739  cd bin
 1740  ls
 1741  cd Linux_x86_64
 1742  pwd
 1743  cd ~
 1744  ls
 1745  emacs .bashrc
 1746  star
 1747  STAR
 1748  source .bashrc
 1749  STAR
 1750  pwd
 1751  ls
 1752  scp axy148@pegasus.ccs.miami.edu:~/Script_bash/RunSTAR.sh ./Code/
 1753  emacs ~/Code/RunSTAR.sh
 1754  ls /media/
 1755  ls /media/DATA/
 1756  ls /media/DATA/hg38
 1757  ls /media/DATA/hg38/Homo_sapiens
 1758  ls /media/DATA/hg38/Homo_sapiens/UCSC
 1759  ls /media/DATA/hg38/Homo_sapiens/UCSC/hg38
 1760  ls /media/DATA/hg38/Homo_sapiens/UCSC/hg38/Sequence/
 1761  mkdir /media/DATA/hg38/Homo_sapiens/UCSC/hg38/Sequence/STARIndex
 1762  emacs ~/Code/RunSTAR.sh
 1763  sh ~/Code/RunSTAR.sh&
 1764  emacs ~/Code/RunSTAR.sh
 1765  STAR
 1766  emacs ~/Code/RunSTAR.sh
 1767  ls /media/
 1768  ls /media/DATA/
 1769  less ~/Code/RunSTAR.sh
 1770  ls /media/DATA/hg38/Homo_sapiens/UCSC/hg38/Sequence/STARIndex
 1771  ls /media/DATA/hg38/Homo_sapiens/UCSC/hg38/Sequence/
 1772  ls /media/DATA/hg38/Homo_sapiens/UCSC/hg38/Sequence/STARIndex/
 1773  ls -alh /media/DATA/hg38/Homo_sapiens/UCSC/hg38/Sequence/STARIndex/
 1774  ls -alh /media/DATA/
 1775  ls -alh /media/DATA/Koji/
 1776  less ~/Code/RunSTAR.sh
 1777  ls -l ~/Code/Com*
 1778  history > ~/Code/CommandHistory27.txt
 1779  exit
 1780  ls 
 1781  STAR
 1782  ls /DATA/
 1783  ls
 1784  ls /media/DATA/
 1785  ls /media/DATA/hg38
 1786  ls /media/DATA/hg38/Homo_sapiens
 1787  ls /media/DATA/hg38/Homo_sapiens/UCSC/
 1788  ls /media/DATA/hg38/Homo_sapiens/UCSC/hg38
 1789  ls /media/DATA/hg38/Homo_sapiens/UCSC/hg38/Sequence/
 1790  ls ~/
 1791  ls /media/
 1792  ls /media/DATA/
 1793  ls /media/DATA/hg38
 1794  ls /media/DATA/hg38/Homo_sapiens
 1795  ls /media/DATA/hg38/Homo_sapiens/UCSC/
 1796  ls /media/DATA/hg38/Homo_sapiens/UCSC/hg38/
 1797  ls /media/DATA/hg38/Homo_sapiens/UCSC/hg38/Sequence/
 1798  ls /media/DATA/hg38/Homo_sapiens/UCSC/hg38/Sequence/STARIndex/
 1799  ls -alh /media/DATA/hg38/Homo_sapiens/UCSC/hg38/Sequence/STARIndex/
 1800  du -sh /media/DATA/hg38/Homo_sapiens/UCSC/hg38/Sequence/STARIndex/
 1801  cd /media/DATA/hg38/Homo_sapiens/UCSC/hg38/Sequence/STARIndex/
 1802  ls
 1803  ls -alh 
 1804  ls
 1805  cd ..
 1806  ls
 1807  scp -r STARIndex/ axy148@pegasus.ccs.miami.edu:~/Homo_sapiens/Homo_sapiens/UCSC/hg38/Sequence/
 1808  du -sh /media/DATA/hg38/Homo_sapiens/UCSC/hg38/Sequence/STARIndex/ls
 1809  ls
 1810  pwd
 1811  ls
 1812  pwd
 1813  ls
 1814  ls ~/Code/Com*
 1815  ls -alh ~/Code/Com*
 1816  history > ~/Code/CommandHistory28.txt
 1817  exit
 1818  ls
 1819  cd Result/
 1820  ls
 1821  cd Jianping/
 1822  ls
 1823  emacs Sample_9.txt
 1824  emacs Sample_9_2.txt
 1825  pwd
 1826  ls
 1827  ls ~/Code/Com*
 1828  ls -alh ~/Code/Com*
 1829  history > ~/Code/CommandHistory29.txt
 1830  exit
 1831  ls
 1832  ls Result/
 1833  ls Result/Jianping/
 1834  cd Result/Jianping/
 1835  ls
 1836  less Sample_9_2.txt
 1837  wc -l *.count
 1838  ls
 1839  ls ~/Code/C*
 1840  history > ~/Code/CommandHistory30.txt
 1841  exit
 1842  Filezilla
 1843  ls ~/Downloads/
 1844  ls -alh ~/Downloads/*.gz
 1845  df -h
 1846  ls /media/DATA/
 1847  ls -alh /media/DATA/
 1848  pwd
 1849  ls
 1850  ls *.R
 1851  mv *.R ~/Code/
 1852  ls
 1853  ls *.sh
 1854  mv *.sh ~/Code/
 1855  Lls
 1856  ls
 1857  ls *.py
 1858  mv *.py ~/Code/
 1859  ls
 1860  mv  README.txt ~/Code/
 1861  mv  README.md ~/Code/
 1862  ls
 1863  mv hg38.2bit /media/DATA/
 1864  ls
 1865  mv Call_UCSC.txt Bw_Files/
 1866  ls
 1867  less examples.desktop
 1868  cd /media
 1869  ls
 1870  cd DATA
 1871  ls
 1872  cd hg38
 1873  ls
 1874  tar xvzf Homo_sapiens_UCSC_hg38.tar.gz
 1875  ls
 1876  cd ~
 1877  ls
 1878  cd Result/
 1879  ls
 1880  ls Jianping/
 1881  less Sample_9.txt
 1882  ls
 1883  cd Jianping/
 1884  ls
 1885  less Sample_9.txt
 1886  cd ..
 1887  ls
 1888  cd ..
 1889  ls
 1890  cd Code
 1891  ls
 1892  ls*.R
 1893  ls *.R
 1894  less Run_DESeq2.R 
 1895  less DESeq2_modified_with_istallation.R
 1896  less DESeq2.R 
 1897  rstudio
 1898  history | grep Rscript
 1899  history
 1900  ls ~/Code/C*
 1901  history > 
 1902  ls ~/Code/C*
 1903  ls -l ~/Code/C*
 1904  ls ~/Code/Com*
 1905  ls -l ~/Code/Com*
 1906  history > ~/Code/CommandHistory31.txt
 1907  exit
 1908  ls ~/Downloads/
 1909  ls -alh ~/Downloads/
 1910  ls -alh ~/Downloads/*.gz
 1911  ls /media/
 1912  ls /media/DATA/
 1913  ls -alh /media/DATA/*.gz
 1914  cd /media/DATA/
 1915  ls
 1916  du -sh
 1917  df
 1918  df -h
 1919  du
 1920  du -h
 1921  du -sh
 1922  du -h
 1923  ls
 1924  ls -l *gz
 1925  ls -lh *gz
 1926  ls
 1927  cd hg38/
 1928  ls
 1929  scp Homo_sapiens_UCSC_hg38.tar.gz axy148@pegasus.ccs.miami.edu:~/Homo_sapiens
 1930  scp Homo_sapiens_UCSC_hg38.tar.gz axy148@pegasus.ccs.miami.edu:~/Homo_sapiens/
 1931  du -sh
 1932  pwd
 1933  ls
 1934  cd ~
 1935  ls
 1936  cd Code
 1937  ls
 1938  ls *.R
 1939  git status
 1940  git staus
 1941  git status
 1942  git add Run_DESeq2.R
 1943  git commit -m "Update run DESeq2"
 1944  git pull --rebase
 1945  git push origin master
 1946  pwd
 1947  ssh axy148@pegasus.ccs.miami.edu
 1948  ls
 1949  ls ~/Code/C*
 1950  ls -l ~/Code/C*
 1951  history > ~/Code/CommandHistory31.txt
 1952  exit
 1953  sudo apt-get clean
 1954  ls
 1955  exit
 1956  down vote
 1957  You can also enter:
 1958  ssh:////bbcresfs.ad.med.miami.edu/Bioinformatics$/ -U cgcent/axy148
 1959  ssh://bbcresfs.ad.med.miami.edu/Bioinformatics$/ -U cgcent/axy148
 1960  ssh://bbcresfs.ad.med.miami.edu/Bioinformatics$ -U cgcent/axy148
 1961  ssh://bbcresfs.ad.med.miami.edu -U cgcent/axy148
 1962  exit
 1963  nautilus-connect-server
 1964  smbclient -L //bbcresfs.ad.med.miami.edu/Bioinformatics$/ -U cgcent/axy148
 1965  //bbcresfs.ad.med.miami.edu/Bioinformatics$/
 1966  ssh //bbcresfs.ad.med.miami.edu/Bioinformatics$/
 1967  ssh://bbcresfs.ad.med.miami.edu/Bioinformatics$/
 1968  exit
 1969  smb://bbcresfs.ad.med.miami.edu/Bioinformatics$/
 1970  man smb
 1971  smb
 1972  smbclient
 1973  nautilus-connect-server
 1974  exit
 1975  ls /media
 1976  mkdir /media/H_driver
 1977  sudo mkdir /media/H_driver
 1978  mount -t cifs smb://bbcresfs.ad.med.miami.edu/Bioinformatics$/ /media/H_driver -o username=axy148 password=Dog#19700908,domain=cgcent
 1979  sudo mount -t cifs smb://bbcresfs.ad.med.miami.edu/Bioinformatics$/ /media/H_driver -o username=axy148 password=Dog#19700908,domain=cgcent
 1980  ls /media/
 1981  ls /media/H_driver/
 1982  cifs
 1983  sudo apt-get install cifs-utils
 1984  sudo mount -t cifs smb://bbcresfs.ad.med.miami.edu/Bioinformatics$/ /media/H_driver -o username=axy148 password=Dog#19700908,domain=cgcent
 1985  exit
 1986  ls ~/Cdoe/
 1987  ls ~/Code/
 1988  ls ~/Code/Comma*
 1989  grep sort  ~/Code/CommandHistory*
 1990  ls ~/Code
 1991  ls ~/Code/ConvertBed2Bw.sh
 1992  emacs ~/Code/ConvertBed2Bw.sh
 1993  grep mtgr1  ~/Code/CommandHistory*
 1994  ls
 1995  locate eto2_peaks.bed
 1996  locate / eto2_peaks.bed
 1997  grep mtgr1  ~/Code/CommandHistory*
 1998  ls /media/DATA/Chip_Seq/
 1999  ls /media/DATA/Chip_Seq/*.bed
 2000  ls /media/DATA/
 2001  ls /media/DATA/ChrisWilliams/
 2002  ls /media/DATA/ChrisWilliams/ChIPseq/*.bed
 2003  awk '{printf "%s\t%d\t%d\t%2.3f\n" , $1,$2,$3,$5}' /media/DATA/ChrisWilliams/ChIPseq/eto2_peaks.bed > /media/DATA/ChrisWilliams/ChIPseq/eto2_peaks.bedgraph
 2004  /home/aiminyan/kentUtils/bin/linux.x86_64/bedGraphToBigWig mtgr1_peaks.bed.bedgraph ../../Data_Chip-Seq/mm10.genome mtgr1_peaks.bw
 2005  /home/aiminyan/kentUtils/bin/linux.x86_64/bedGraphToBigWig /media/DATA/ChrisWilliams/ChIPseq/eto2_peaks.bedgraph ./Data_Chip-Seq/mm10.genome /media/DATA/ChrisWilliams/ChIPseq/eto2_peaks.bw
 2006  ls 
 2007  ls ./Mus_musculus/
 2008  ls ./Mus_musculus/UCSC
 2009  /home/aiminyan/kentUtils/bin/linux.x86_64/bedGraphToBigWig /media/DATA/ChrisWilliams/ChIPseq/eto2_peaks.bedgraph ./Mus_musculus/UCSC/mm10.genome /media/DATA/ChrisWilliams/ChIPseq/eto2_peaks.bw
 2010  vi ./Mus_musculus/UCSC/mm10.genome
 2011  cp ./Mus_musculus/UCSC/mm10.genome ./Mus_musculus/UCSC/mm10_re_header.genome
 2012  vi ./Mus_musculus/UCSC/mm10_re_header.genome
 2013  /home/aiminyan/kentUtils/bin/linux.x86_64/bedGraphToBigWig /media/DATA/ChrisWilliams/ChIPseq/eto2_peaks.bedgraph ./Mus_musculus/UCSC/mm10_re_header.genome /media/DATA/ChrisWilliams/ChIPseq/eto2_peaks.bw
 2014  history > ~/Code/ConvertBed2Bw_12_22_2015.sh
