#FPKM_count_2.py -i "/media/H_driver/2016/Ramin_azhang/for_bioinfo_core/RNA_seq/""2016-02-10-emp1_WT".alignments.bam -r /media/H_driver/2016/Ramin_azhang/Annotation/hg19_gene_pos_neg_downstream_sorted_3.bed -d '+-,-+' -q 0 -s 1 -o "/media/H_driver/2016/Ramin_azhang/for_bioinfo_core/RNA_seq/Results4NewData/""2016-02-10-emp1_WT".gene.downstream.count.plus.minus.and.minus.plus.2.txt

FPKM_count.py -i "/media/H_driver/2016/Ramin_azhang/for_bioinfo_core/RNA_seq/""2016-02-10-emp1_WT".alignments.bam -r /media/H_driver/2016/Ramin_azhang/Annotation/hg19_gene_pos_neg_downstream_sorted_4.bed -d '++,--' -q 0 -s 1 -o "/media/H_driver/2016/Ramin_azhang/for_bioinfo_core/RNA_seq/Results4NewData/""2016-02-10-emp1_WT".gene.downstream.count.plus.plus.and.minus.minus.4.DoGs

FPKM_count.py -i "/media/H_driver/2016/Ramin_azhang/for_bioinfo_core/RNA_seq/""2016-02-10-emp1_WT".alignments.bam -r /media/H_driver/2016/Ramin_azhang/Annotation/hg19_gene_pos_neg_downstream_sorted_4.bed -d '+-,-+' -q 0 -s 1 -o "/media/H_driver/2016/Ramin_azhang/for_bioinfo_core/RNA_seq/Results4NewData/""2016-02-10-emp1_WT".gene.downstream.count.plus.minus.and.minus.plus.4.DoGs

FPKM_count.py -i "/media/H_driver/2016/Ramin_azhang/for_bioinfo_core/RNA_seq/""2016-02-10-emp1_WT".alignments.bam -r /media/H_driver/2016/Ramin_azhang/Annotation/hg19_gene_pos_neg_plus_downstream_sorted_7.bed -d '++,--' -q 0 -s 1 -o "/media/H_driver/2016/Ramin_azhang/for_bioinfo_core/RNA_seq/Results4NewData/""2016-02-10-emp1_WT".gene.downstream.count.plus.plus.and.minus.minus.4.gene.plus.DoGs

FPKM_count.py -i "/media/H_driver/2016/Ramin_azhang/for_bioinfo_core/RNA_seq/""2016-02-10-emp1_WT".alignments.bam -r /media/H_driver/2016/Ramin_azhang/Annotation/hg19_gene_pos_neg_plus_downstream_sorted_7.bed -d '+-,-+' -q 0 -s 1 -o "/media/H_driver/2016/Ramin_azhang/for_bioinfo_core/RNA_seq/Results4NewData/""2016-02-10-emp1_WT".gene.downstream.count.plus.minus.and.minus.plus.4.gene.plus.DoGs

#FPKM_count.py -i "/media/H_driver/2016/Ramin_azhang/for_bioinfo_core/RNA_seq/""2016-02-10-emp1_WT".alignments.bam -r /media/H_driver/2016/Ramin_azhang/Annotation/hg19_gene_pos_neg_downstream_sorted_5.bed -d '++,--' -q 0 -s 1 -o "/media/H_driver/2016/Ramin_azhang/for_bioinfo_core/RNA_seq/Results4NewData/""2016-02-10-emp1_WT".gene.downstream.count.plus.plus.and.minus.minus.4

#FPKM_count.py -i "/media/H_driver/2016/Ramin_azhang/for_bioinfo_core/RNA_seq/""2016-02-10-emp1_WT".alignments.bam -r /media/H_driver/2016/Ramin_azhang/Annotation/hg19_gene_pos_neg_downstream_sorted_5.bed -d '+-,-+' -q 0 -s 1 -o "/media/H_driver/2016/Ramin_azhang/for_bioinfo_core/RNA_seq/Results4NewData/""2016-02-10-emp1_WT".gene.downstream.count.plus.minus.and.minus.plus.4

